dask_sql/physical/rel/logical/project.py

```diff
@@ -46,8 +46,11 @@ def convert(self, rel: "LogicalPlan", context: "dask_sql.Context") -> DataContai
             # shortcut: if we have a column already, there is no need to re-assign it again
             # this is only the case if the expr is a RexInputRef
             if expr.getRexType() == RexType.Reference:
-                index = expr.getIndex()
-                backend_column_name = cc.get_backend_by_frontend_index(index)
+                try:
```

Comment:
I haven't looked into this deeper but do we know why our `expr.getIndex()` call fails for cases like these? It seems like the schema does have `1b` but somehow we don't pick it up as expected.

Reply:
You're right, thanks! Looking further into our [getIndex()](https://github.com/dask-contrib/dask-sql/blob/f895346cb878e384410aa1ce2fd3bb3161d560e1/dask_planner/src/expression.rs#L173) function, it works as expected until we get to https://github.com/dask-contrib/dask-sql/blob/f895346cb878e384410aa1ce2fd3bb3161d560e1/dask_planner/src/expression.rs#L184

At first, I thought the issue was DataFusion's [index_of_column()](https://github.com/apache/arrow-datafusion/blob/f6e49ac7a027abb95d8b7fa755502dfa7d53c21c/datafusion/common/src/dfschema.rs#L238) function not behaving as expected. Actually, it's the [from_qualified_name()](https://github.com/apache/arrow-datafusion/blob/ec24724ea1130909402c64d5651a5815dcaf8285/datafusion/common/src/column.rs#L54) function, which we can create a workaround for on the Dask-SQL side. Just pushed new changes, let me know what you think.

Reply:
Thanks for looking into this. I dug into this a bit deeper and it turns out that `Column::from_qualified_name(name)` calls sqlparser's tokenizer to tokenize the qualified name.
For cases like these `table_pq.a` it gets tokenized to 
```
[
  Word(
    Word { value: "table_pq", quote_style: None, keyword: NoKeyword }), 
    Period, 
    Word(Word { value: "a", quote_style: None, keyword: NoKeyword }
  )
]
```
while for cases like these `table_pq.1b` it's tokenized to 
```
[
  Word(
    Word { value: "table_pq", quote_style: None, keyword: NoKeyword }), 
    Number(".1", false), 
    Word(Word { value: "b", quote_style: None, keyword: NoKeyword }
  )
]
```

I'm not sure if it's a bug on sqlparser's end or if datafusion should separate the qualified name differently.

Reply:
For now, I opened an issue https://github.com/apache/arrow-datafusion/issues/5426 about this on the DataFusion side.